tour/main.js

```diff
@@ -53,19 +53,31 @@ function observeBlocks(myFunction) {
   });
 }
 
-const toggleAutoTour = (SCRIPT_API) => {
+const toggleAutoTour = (SCRIPT_API, setTo = !showAutoTour) => new Promise((resolve, reject) => {
   onAuthenticated(async () => {
     const token = await window.auth0Client.getTokenSilently();
     const headers = { authorization: `bearer ${token}` };
 
     await fetch(`${SCRIPT_API}/userSettings`, {
```

Comment:
Maybe you can replace this with a generic function that takes newDate and updates the userSettings and the cache